week4/yoonseo/app.ts

```diff
@@ -0,0 +1,115 @@
+const todoListEl = document.getElementById("todoList") as HTMLElement;
+const todoInputEl = document.getElementById("todoInput") as HTMLInputElement;
+
+const API_URL = "http://localhost:8080/todos";
+
+interface Todo {
+    id: number;
+    title: string;
+    createdAt: string;
+    completed: boolean;
+}
+
+interface Todos {
+    todos: Todo[];
+}
+
+const fetchTodos = async (): Promise<Todo[]> => {
+    const response = await fetch(API_URL);
+    const data: Todos = await response.json();
+    return data.todos;  //Todos 인터페이스의 todos 배열을 반환
+};
+
+//페이지가 로드될 때 fetchTodos 함수를 호출하고, 그 결과를 renderTodo에 전달
+document.onload = async () => {
+    const todos = await fetchTodos();
+    renderTodo(todos);
+}
+ 
+//4주차 mission: typescript로 바꾸고, async와 await 사용하는 방법으로 바꾸기
+//document onLoad eventListener : onLoad 될 때 fetchTodos를 실행해서 renderTodo에 넘기는 것 - 좋은 방식
+
+const updateTodo = async (todoId: number, originalTitle: string) => {
+    const todoItem = document.querySelector(`#todo-${todoId}`) as Element;
+    todoItem.innerHTML = "";
+    // mission
+    const inputTitle = document.createElement("input");
+    inputTitle.type = "text";
+    inputTitle.value = originalTitle;
+    inputTitle.className = "updateInput";
+
+    todoItem.appendChild(inputTitle);
+
+    inputTitle.focus();
+
+    inputTitle.addEventListener("keypress", async (event) => {
+        if(event.key == "Enter" && inputTitle.value.trim() !== "" && inputTitle.value !== originalTitle){
```

Comment:
if문 하나에 여러개의 조건을 넣는것은 좋지 않습니다!
코드 가독성이 떨어지고 각 조건의 의미를 한눈에 파악하기 어려워집니다.
```suggestion
        if(event.key == "Enter"){
          if(inputTitle.value.trim() === "" || inputTitle.value === originalTitle){
            return;
          }
          // 나머지...
        }
```
이렇게 분리하면 enter 이벤트라는 의미가 처음 보여지고 그 다음 조건이 들어오게 됩니다.